src/components/shared/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,119 @@
+'use client'
+
+import React, { useState, useRef, ChangeEvent, FormEvent } from 'react';
+import { FormControl, makeStyles } from '@mui/material';
+import StorefrontIcon from '@mui/icons-material/Storefront';
+import Inventory2Icon from '@mui/icons-material/Inventory2';
+import SearchIcon from '@mui/icons-material/Search';
+import IconButton from '@mui/material/IconButton';
+import TextField from '@mui/material/TextField';
+import { useTranslation } from 'react-i18next';
+import Snackbar from '@mui/material/Snackbar';
+import './SearchBar.scss';
+
+const SearchBar: React.FC = () => {
+  const [searchBarValue, setSearchBarValue] = useState('');
+  const [isBusinessSearchType, setIsBusinessSearchType] = useState(true);
+  const [message, setMessage] = useState('');
+  const { t } = useTranslation();
+
+  const handleSearchBarChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setSearchBarValue(event.target.value);
+  };
+  
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleIconClick = () => {
+    // Move the placeholder text up
+    if(inputRef.current) {
+      inputRef.current.focus();
+    }
+  };
+
+  const fetchSearchResults = async (query: string, searchType: string):Promise<any[]> => {
+    // fetch logic to be defined here
+    // For now, it returns an empty array
+    return new Promise((resolve) => resolve([]));
+  };
+
+  const setSearchQuery = (query: string, searchType: string) => {
+    // setSearchQuery logic here
+  };
+  
+  const handleSubmitSearch = async (event: FormEvent) => {
+    event.preventDefault();
+    const query = searchBarValue;
+    if (query) {
+      const searchType = isBusinessSearchType ? 'business' : 'product';
+      console.info(`Search query emitted for ${searchType}: `, query);
```

Comment:
We'll eventually need to replace these console logging with a logging framework; discussed **winston** earlier.